client/service/src/client/call_executor.rs

```diff
@@ -81,15 +83,23 @@ where
 		Block: BlockT,
 		B: backend::Backend<Block>,
 	{
-		let code = self.wasm_override
+		let code = if let Some(d) = self.wasm_override
 			.as_ref()
 			.map::<sp_blockchain::Result<Option<RuntimeCode>>, _>(|o| {
 				let spec = self.runtime_version(id)?.spec_version;
```

Comment:
dq: is it intentional to return `Err` if runtime version couldn't be found?

Reply:
Yes, otherwise we can't know what runtime to use to execute that block. I suppose we could also coerce it to an Option but that will inevitably lead to silent failures that will be difficult to debug I think

Reply:
Yes this is done by intention. Substrate runtimes need to provide a version. This is a fundamental requirement of substrate.